### PR TITLE
fix: trim root folder `/` when calculating relative paths

### PR DESCRIPTION
**Pull request by peterroe**
### 🔗 Linked issue

resolves https://github.com/unjs/pathe/issues/126

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

The `split` method will generate redundant `""` when pass a root path

https://github.com/unjs/pathe/blob/055f50a6f1131f4e5c56cf259dd8816168fba329/src/path.ts#L213-L215

```ts
"/".split("/") //=> ["", ""] 
// But expected [""]
```

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -61,3 +61,7 @@ export function filename(path: string) {
 function _compareAliases(a: string, b: string) {
   return b.split("/").length - a.split("/").length;
 }
+
+export function _isRootPath(path: string) {
+  return path === "/" || /^\/[a-z]:$/i.test(path);
```
Comment (pi0) [inline src/utils.ts:66]: MAybe call it _isAbsolutePath? Also windows drive would be nice if we explicitly check for `a-zA-Z` for drive path casing.

Comment (peterroe) [reply src/utils.ts]: Thanks. That is a very nice suggestion, I have optimized the code.

**Review by pi0 — APPROVED** [review]
LGTM (have to test locally for possible perf improvement tough)

File: src/path.ts [hunk 2]
```diff
@@ -213,6 +213,12 @@ export const extname: typeof path.extname = function (p) {
 export const relative: typeof path.relative = function (from, to) {
   const _from = resolve(from).split("/");
   const _to = resolve(to).split("/");
+  if (_isAbsolutePath(resolve(from))) {
```
Comment (danielroe) [inline src/path.ts:216]: Can we avoid the double call to resolve?

**Comment by peterroe** [comment]
All set 👀

**Comment by peterroe** [comment]
👀 Maybe we should fix this in time? cc @danielroe

**Review by danielroe — APPROVED** [review]

**Pull request merged**